iati/validator.py

```diff
@@ -331,17 +336,20 @@ def _check_rules(dataset, ruleset):
         dataset (iati.core.data.Dataset): The Dataset to check Ruleset conformance with.
         ruleset (iati.code.Ruleset): The Ruleset to check conformance with.
 
-
     Returns:
         iati.validator.ValidationErrorLog: A log of the errors that occurred.
 
     """
     error_log = ValidationErrorLog()
 
-    if not ruleset.is_valid_for(dataset):
-        error = ValidationError('err-ruleset-conformance-fail', locals())
+    for rule in ruleset.rules:
+        if rule.is_valid_for(dataset) is None:
+            error = ValidationError('warn-rule-skipped', locals())
+            error_log.add(error)
 
-        error_log.add(error)
+        elif rule.is_valid_for(dataset) is False:
```

Comment:
Duplicate call to `rule.is_valid_for()` - value can be assigned to a variable so it's only called once.